src/builder.rs

```diff
@@ -552,19 +643,110 @@ impl Builder {
             self.outputs,
         )
     }
+
+    /// Builds a bundle containing the given spent notes and outputs along with their
+    /// metadata, for inclusion in a PCZT.
+    pub fn build_for_pczt(
+        self,
+        rng: impl RngCore,
+    ) -> Result<(crate::pczt::Bundle, BundleMetadata), BuildError> {
+        build_bundle(
+            rng,
+            self.anchor,
+            self.bundle_type,
+            self.spends,
+            self.outputs,
+            |pre_actions, flags, value_sum, bundle_meta, mut rng| {
+                // Create the actions.
+                let actions = pre_actions
+                    .into_iter()
+                    .map(|a| a.build_for_pczt(&mut rng))
+                    .collect::<Vec<_>>();
+
+                Ok((
+                    crate::pczt::Bundle {
+                        actions,
+                        flags,
+                        value_sum,
+                        anchor: self.anchor,
+                        zkproof: None,
+                        bsk: None,
+                    },
+                    bundle_meta,
+                ))
+            },
+        )
+    }
 }
 
 /// Builds a bundle containing the given spent notes and outputs.
 ///
 /// The returned bundle will have no proof or signatures; these can be applied with
 /// [`Bundle::create_proof`] and [`Bundle::apply_signatures`] respectively.
 pub fn bundle<V: TryFrom<i64>>(
-    mut rng: impl RngCore,
+    rng: impl RngCore,
     anchor: Anchor,
     bundle_type: BundleType,
     spends: Vec<SpendInfo>,
     outputs: Vec<OutputInfo>,
 ) -> Result<Option<(UnauthorizedBundle<V>, BundleMetadata)>, BuildError> {
+    build_bundle(
+        rng,
+        anchor,
+        bundle_type,
+        spends,
+        outputs,
+        |pre_actions, flags, value_balance, bundle_meta, mut rng| {
+            let result_value_balance: V = i64::try_from(value_balance)
+                .map_err(BuildError::ValueSum)
+                .and_then(|i| {
+                    V::try_from(i).map_err(|_| BuildError::ValueSum(value::OverflowError))
+                })?;
+
+            // Compute the transaction binding signing key.
+            let bsk = pre_actions
+                .iter()
+                .map(|a| &a.rcv)
+                .sum::<ValueCommitTrapdoor>()
+                .into_bsk();
+
+            // Create the actions.
+            let (actions, circuits): (Vec<_>, Vec<_>) =
+                pre_actions.into_iter().map(|a| a.build(&mut rng)).unzip();
+
+            // Verify that bsk and bvk are consistent.
+            let bvk = (actions.iter().map(|a| a.cv_net()).sum::<ValueCommitment>()
+                - ValueCommitment::derive(value_balance, ValueCommitTrapdoor::zero()))
+            .into_bvk();
+            assert_eq!(redpallas::VerificationKey::from(&bsk), bvk);
```

Comment:
Internal consistency check; there's a straight line between where we sample `rcv` and compute `cv_net`, and here.

Note how in the PCZT version I turned this check into an error because we're serializing in between and thus there's a chance that a Constructor put bad `cv_net` or `rcv` data into the PCZT that would cause this to fail.